src/common/stream_coder.h

```diff
@@ -0,0 +1,23 @@
+#ifndef STREAM_CODER_H
+#define STREAM_CODER_H
+
+#include "allocator.h"
+#include <nan.h>
+
+using namespace v8;
+
+class StreamCoder : public Nan::ObjectWrap {
+  public:
+    const uint8_t* next_in;
+    size_t available_in;
```

Comment:
Back from lunch!

Well we should strive to for what's best of course.
I gave a quick look at `BrotliEncoderCompressStream()` and it seems to handle filling the block for us, so we should be able to remove that part of our code.
https://github.com/google/brotli/blob/master/enc/encode.c#L1510-L1519